### ISSUE TITLE
Improvement to throw ExceptionInfo output 

### ISSUE THREAD

**Issue by GreshamDanielStephens** [babashka/babashka#730]
Currently when an ExceptionInfo is thrown, the data map details are lost in the output from the error. It would be nice to have the data presented.
To recreate:
```
(def some-map {:a 1 :b 2 :c 3})
(throw (ex-info "ABC" some-map))
```

Current output:
```
----- Error --------------------------------------------------------------------
Type:     clojure.lang.ExceptionInfo
Message:  ABC
Location: path/x/y/z.clj----- Context ------------------------------------------------------------------
245: 
246: 
247: 
248: 
249: (throw (ex-info "ABC" some-map))
     ^--- ABC
```

Improvement would be to add data into the output as:
```
----- Error --------------------------------------------------------------------
Type:     clojure.lang.ExceptionInfo
Message:  ABC
Data: XYZ
```

**Comment by GreshamDanielStephens** [comment]
I started having a look at this but I think it's slightly more involved than I expected.
I set up a new test, doing:
```
(try (tu/bb nil "
(defn foo [] (throw (ex-info \"ex-info message\" {:some :data})))
(foo)")
                    (catch Exception e (ex-message e)))
```

Interestingly under the hood this produces an ExceptionInfo with the data:
```
{:type :sci/error, :line 2, :column 14, :message \"ex-info message\", :sci.impl/callstack #object[clojure.lang.Delay 0x7fe33ae0 {:status :ready, :val ((#'user/foo))}], :file \"<expr>\", :locals {}, :some :data}
```

So as can be seen the ex-data from the code actually gets merged in to the sci exception ex-data at the top level.
I thought that looked a bit worrying so I tried doing `(throw (ex-info \"ex-info message\" {:column :data}))` instead and sure enough we get a weird exception, which is actually a babashka bug because column gets overwritten in the sci exception with a non-int value.
```
class clojure.lang.Keyword cannot be cast to class java.lang.Number (clojure.lang.Keyword is in unnamed module of loader 'app'; java.lang.Number is in module java.base of loader 'bootstrap')
```

I haven't got too much further than that but my initial assumption from the `:type :sci/error` is that this will first require a bug fix in sci, to nest the 'inherited' ex-data down one level I think, that sounds like it might be a breaking change but I will start looking at where that fix might live when I next get a chance.

**Comment by borkdude** [comment]
@GreshamDanielStephens I think it might be better if we didn't merge the ex-data from the cause exception if it's not a :sci/error, but assoc it like `:ex-data ex-data` so we preserve the original better.

**Comment by borkdude** [comment]
@GreshamDanielStephens You are close. It's probably this one: 

https://github.com/borkdude/sci/blob/002cc9418ad3a28b2f925177afd6ed5065cca6e6/src/sci/impl/utils.cljc#L134

**Comment by borkdude** [comment]
But wait a minute. We also preserve the entire exception in the ex-cause field of the ex-data. Can't we use that?

**Comment by borkdude** [comment]
@GreshamDanielStephens If you add this:

``` 
(prn (some-> e ex-cause ex-data))
``` 

to `error-handler` in `babashka.impl.error-handler`, you will see:

``` 
user=> (main/main "(throw (ex-info \"foo\" {:column 1337}))")
{:column 1337}
----- Error --------------------------------------------------------------------
Type:     clojure.lang.ExceptionInfo
Message:  foo
Location: <expr>:1:1337

----- Context ------------------------------------------------------------------
1: (throw (ex-info "foo" {:column 1337}))
```

**Comment by borkdude** [comment]
Still I think it's better to separate the data from the original exception and not merge it. So welcome to PR this in sci as well.

**Comment by GreshamDanielStephens** [comment]
Yeah I agree with that, do you have any preference for if the fix here is based on `-> ex-cause ex-data` or if I should look at the sci improvement first and then just use `:ex-data` here once it exists.

**Comment by borkdude** [comment]
I think it makes sense to do the sci improvement first.

**Comment by borkdude** [comment]
@GreshamDanielStephens I think this issue can now be implemented (if you still want to) now that sci has the proper support for it.

**Comment by borkdude** [comment]
Fixed by https://github.com/babashka/babashka/pull/854
